got-no-way.py

```
@@ -0,0 +1,42 @@
+import os
+
+
+def get_path() -> str:
+    return input('Please enter path -> ')
+
+
+def get_file_list(path_to_file: str) -> list:
+    directory = None
+    try:
+        directory = os.listdir(path_to_file)
+    except:
+        show_error('Error opening file.')
```

The point about separating IO from logic is not the exclude in to a function, but to not handle output inside functions that use logic. I would read about what exception and let it be thrown if the file is missing. 
read this - 
https://softwareengineering.stackexchange.com/questions/139171/check-first-vs-exception-handling